src/movies.js

```diff
@@ -1,16 +1,132 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(movies){
+       //code
+       let answer = movies.map((movie)=>{
+           return movie.director;
+       })
+       return answer;
+}
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-
+function howManyMovies(movies){
+    let dramaAndSteven;
+    let answer = movies.filter((movie)=>{
+        for(let i=0; i<movie.genre.length; i++){
+            dramaAndSteven = movie.genre[i] == "Drama" && movie.director == 'Steven Spielberg';
+          if(dramaAndSteven){break;}
+        }
+        return dramaAndSteven;
+    })
+    console.log(answer);
+    return answer.length;
+}
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(movies){
+    if(movies==0){return 0}
+    else{
+        let avrg;
+        let answer = movies.reduce((acc, elem)=>{
+            return acc + (elem.rate || 0);
+        },0)
+        avrg = answer/movies.length;
+        return parseFloat(avrg.toFixed(2));
+   }
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-
+function dramaMoviesRate(movies){
+    let dmSp;
+    let dramaMovies = movies.filter((movie)=>{
+        for(let i=0; i<movie.genre.length; i++){
+        dmSp = movie.genre[i] == "Drama";
+          if(dmSp){break;}
+        }
+        return dmSp;
+    })
+    return ratesAverage(dramaMovies);
+  
+}
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-
+function orderByYear(movies){
+    let cloneMovie = JSON.parse(JSON.stringify(movies));
+    let byYear = cloneMovie.sort((a,b)=>{
+        if(a.year>b.year){return 1;}
+        else if(a.year<b.year){return -1;}
+        else{
+            if(a.title>b.title){return 1;}
+            else if(a.title<b.title){return -1;}
+            else{return 0;}
+        }
+    });
+    return byYear;
+}
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
+function orderAlphabetically(movies){
+    let cloneMovie = JSON.parse(JSON.stringify(movies));
+    let the20first=[];
+    let titlesInAlphabeticalOrder = cloneMovie.sort((a,b)=>{
```

Comment:
Another tip, since you are using arrow functions, you can research the reduced syntax for the sort method when only used to do a simple sort. You could essentially turn 5 lines into 1 and still look very clean.